### PR TITLE
Removing remaining SLEEF related code

### PR DESCRIPTION
**Pull request by juntuu**
I removed all the remaining SLEEF conditional code that I could find. Including few macros that were only used by the SLEEF code.

This might also address #21.

There are two tests failing on my system (osx) both before and after the changes. I added what information I could gather below, in case those might need some attention at some point.

```
99% tests passed, 2 tests failed out of 401

Total Test time (real) = 339.31 sec

The following tests FAILED:
          7 - g001 (Failed)
        180 - g420stch (SEGFAULT)
Errors while running CTest
FAILED: CMakeFiles/Debug/test.util
```

The first one I could not reproduce in the jconsole, and it also passed when running single threaded, so it might be some race condition between tests.

The second one I traced back to the following snippet:

```ijs
etx      =: 1 : 'u :: (<:@(13!:11)@i.@0: >@{ 9!:8@i.@0:)'  NB. error message from error number -- from test/tsu.ijs:59
'length error' -: ,.&.>/ etx 1 2;3 4 5                     NB. SEGFAULT here
```

and here is where the debugger located the segfault.

```
(lldb) target create "./build/jsrc/Debug/jconsole"
Current executable set to './build/jsrc/Debug/jconsole' (x86_64).
(lldb) r
Process 37931 launched: '.../build/jsrc/Debug/jconsole' (x86_64)
Hello YouTube Viewers, all 22 of you!
   etx      =: 1 : 'u :: (<:@(13!:11)@i.@0: >@{ 9!:8@i.@0:)'
   ,.&.>/ etx 1 2;3 4 5
length error
Process 37931 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0xe000020)
    frame #0: 0x000000010292288d libj.dylib`jtfa(jt=0x0000000101016000, wd=0x00000001006110c0, t=32) at m.c:736:3
   733     fana(np);  // increment the box, possibly turning it to recursive
   734     np=np0;  // advance to next box
   735    };
-> 736    fana(np);  // increment the box, possibly turning it to recursive
   737   } else if(t&(VERB|ADV|CONJ)){V* RESTRICT v=FAV(wd);
   738    // ACV.
   739    fana(v->fgh[0]); fana(v->fgh[1]); fana(v->fgh[2]);
Target 0: (jconsole) stopped.
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0xe000020)
  * frame #0: 0x000000010292288d libj.dylib`jtfa(jt=0x0000000101016000, wd=0x00000001006110c0, t=32) at m.c:736:3
    frame #1: 0x00000001029225fa libj.dylib`jttpop(jt=0x0000000101016000, old=0x000000010101c000) at m.c:862:11
    frame #2: 0x00000001029150bd libj.dylib`jdo(jt=0x0000000101016000, lp=",.&.>/ etx 1 2;3 4 5") at io.c:288:2
    frame #3: 0x0000000102915a35 libj.dylib`JDo(jt=0x0000000101016000, lp=",.&.>/ etx 1 2;3 4 5") at io.c:382:9
    frame #4: 0x0000000100001f5b jconsole`::jedo(sentence=",.&.>/ etx 1 2;3 4 5") at jeload.cpp:44:12
    frame #5: 0x0000000100001dd0 jconsole`main(argc=1, argv=0x00007ffeefbff220) at jconsole.c:293:11
    frame #6: 0x00007fff71606cc9 libdyld.dylib`start + 1
(lldb)
```

### CODE REVIEW

File: jsrc/j.c [hunk 1]
```diff
@@ -129,11 +129,6 @@ F1(jtversq){
  strcat(p, "-");
  sprintf(p + strlen(p), "%i", __GNUC_MINOR__);
 #endif
- 
-#if SLEEF
- strcat(p,"/SLEEF=1");
-#else
- strcat(p,"/SLEEF=0");
-#endif
```
Comment (juntuu) [inline jsrc/j.c:137]: This might address #21

**Comment by juntuu** [comment]
The failing test in the pipeline seems to test memory mapped files based on the comment.

Could not reproduce this locally.

```
The following tests FAILED:
	298 - gmmf (Bus error)
Errors while running CTest
```

**Comment by ahmadbeirkdar** [comment]
> There are two tests failing on my system (osx) both before and after the changes. I added what information I could gather below, in case those might need some attention at some point.
> 
> ```
> 99% tests passed, 2 tests failed out of 401
> 
> Total Test time (real) = 339.31 sec
> 
> The following tests FAILED:
>           7 - g001 (Failed)
>         180 - g420stch (SEGFAULT)
> Errors while running CTest
> FAILED: CMakeFiles/Debug/test.util
> ```
> 

Yup, these two tests are failing on my macOS machine also.

**Comment by codereport** [comment]
If you merge `main`, build should pass at least.

**Review by codereport — APPROVED** [review]
This is :fire: More deleted code the better! :pray:

**Comment by juntuu** [comment]
Merged main, and removed couple more lines that were left unused from the previous changes.

Ps. Could not join the livestream last night, it was a bit too late for me. Seems you had some great progress 👍

**Pull request merged**